hcache/hcache.c

```diff
@@ -380,9 +414,78 @@ void *mutt_hcache_fetch_raw(header_cache_t *hc, const char *key, size_t keylen)
   struct Buffer path = mutt_buffer_make(1024);
 
   keylen = mutt_buffer_printf(&path, "%s%s", hc->folder, key);
-
-  void *blob = ops->fetch(hc->ctx, mutt_b2s(&path), keylen);
+  size_t dlen;
+  void *blob = ops->fetch(hc->ctx, mutt_b2s(&path), keylen, &dlen);
   mutt_buffer_dealloc(&path);
+
+#ifdef USE_HCACHE_COMPRESSION
+  if (blob == NULL)
+    return blob;
+  const char *cm = C_HeaderCacheCompressMethod;
+#endif
+
+#ifdef HAVE_LZ4
+  if (cm && strcmp(cm, "lz4") == 0 && dlen > 10) {
```

Comment:
1. `cm && strcmp(cm, "lz4")` can be simplified as  `mutt_str_strcmp(cm, "lz4")`
2. I'd move the check for dlen before the equality check for the string, with the rationale that if dlen <= 10, we don't event call a function to compare the string

Reply:
3. we put braces on new lines

Reply:
1..3 all done ;)